src/clojure/nrepl/misc.clj

```diff
@@ -106,7 +113,7 @@
       (update :ns str)
       (update :name str)
       (update :protocol str)
-      (update :file #(or (some-> (str %) io/resource str) %))
+      (update :file #(or (some-> % check-filetype str) %))
```

Comment:
I'd replace this entire lambda with a function, so you can extend the function you introduced and name it something like `handle-file-meta` or something like this.

Reply:
Did it, but I think I got it worse :\